ethcore/private-tx/src/lib.rs

```diff
@@ -240,9 +251,8 @@ impl Provider where {
 
 	/// Try to unlock account using stored passwords
 	fn unlock_account(&self, account: &Address) -> bool {
-		let passwords = self.config.passwords.clone();
-		for password in passwords {
-			if let Ok(()) = self.accounts.unlock_account_temporarily(account.clone(), password) {
+		for password in &self.passwords {
+			if let Ok(()) = self.accounts.unlock_account_temporarily(account.clone(), password.clone()) {
```

Comment:
this is a huge security flaw,

signing operation is not atomic: unlock_account unlocks account for one usage. there is no guarantee that one or more other service will not use unlocked accounts in the meantime

Reply:
Reworked